android/app/src/main/java/com/androidstartupperfapp/MainActivity.java

```diff
@@ -12,4 +16,33 @@ public class MainActivity extends ReactActivity {
   protected String getMainComponentName() {
     return "AndroidStartupPerfApp";
   }
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    this.initializeLandingScreen();
+  }
+
+  @Override
+  public void invokeDefaultOnBackPressed() {
+    this.maybeGoBackToLandingScreen();
+  }
+
+  // TODO: Relinquish all of this behavior to a custom navigator.
+
+  private void initializeLandingScreen() {
+    LandingScreenFragment landingScreenFragment = MainActivityScreensManager.createLandingScreenFragment();
+    landingScreenFragment.addAsync(this);
+  }
+
+  private void maybeGoBackToLandingScreen() {
+    LandingScreenFragment landingScreenFragment = MainActivityScreensManager.getLandingScreenFragment();
+    if (landingScreenFragment.isHidden()) {
+      // Reset the React app navigation.
+      InitialRouteNameState.setValue(null);
+
+      // Show the Android landing screen fragment to simulate back button press navigation.
+      landingScreenFragment.showAsync(this);
+    }
+  }
```

Comment:
## Back button navigation to home screen disabled

In order to honor navigation back to home screen, something like this would be needed:

```java
  @Override
  public void invokeDefaultOnBackPressed() {
    boolean alreadyAtLandingScreen = !this.maybeGoBackToLandingScreen();

    if (alreadyAtLandingScreen) {
      // Handle back button in React app to pause it.
      this.getReactInstanceManager().onBackPressed();
      // Handle back button in Android fragment to pause it.
      super.onBackPressed();
    }
  }

  private boolean maybeGoBackToLandingScreen() {
    LandingScreenFragment landingScreenFragment = MainActivityScreensManager.getLandingScreenFragment();

    // Landing screen is currently shown.
    // There's nothing else to "go back" to for the React app navigation.
    if (!landingScreenFragment.isHidden()) {
      return false;
    }

    // Reset the React app navigation.
    InitialRouteNameState.setValue(null);

    // Show the Android landing screen fragment to simulate back button press navigation.
    landingScreenFragment.showAsync(this);
    
    return true;
  }
```